raiden/network/transport/matrix/client.py

```diff
@@ -72,11 +78,13 @@ def get_joined_members(self, force_resync: bool = False) -> List[User]:
     def leave(self) -> None:
         """ Leave the room. Overriding Matrix method to always return error when request. """
         self.client.api.leave_room(self.room_id)
-        del self.client.rooms[self.room_id]
+        self.client.rooms.pop(self.room_id, None)
```

Comment:
Because of a [dummy room object is used](https://github.com/raiden-network/raiden/pull/5709/files#diff-063d1686c23ac92538ea0d0df8face3bR859-R860) to reject invalid invites (#5640)